evalml/tests/component_tests/test_delayed_features_transformer.py

```diff
@@ -0,0 +1,115 @@
+import pandas as pd
+import pytest
+
+from evalml.pipelines import DelayedFeaturesTransformer
+
+
+@pytest.fixture
+def delayed_features_data():
+    X = pd.DataFrame({"feature": range(1, 32)})
+    y = pd.Series(range(1, 32))
+    return X, y
+
+
+def test_delayed_features_transformer_init():
+
+    delayed_features = DelayedFeaturesTransformer(max_delay=4, random_state=1)
+    assert delayed_features.parameters == {"max_delay": 4}
+
+
+def test_lagged_feature_extractor_maxdelay3_gap1(delayed_features_data):
```

Comment:
These tests could be condensed into one test with `parametrize` but I think that being explicit here makes it easier to see that the output matches what's in the design doc.